### PR TITLE
`Development`: Add rules for pull request titles into the documentation

### PR DESCRIPTION
**Pull request by MDK5**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).

### Motivation, Context and Description
In order to keep the PR titles unified guidelines were proposed in slack. This PR manifests those in the documentation by appending them to the already existing PR workflow page.

View the changed documentation using the [link from the GitHub actions](https://artemis-platform--4259.org.readthedocs.build/en/4259/dev/guidelines/development-process/).

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

File: docs/dev/guidelines/development-process.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+3. Review
+---------
+
+Developer
+---------
+* Organize or join a testing session. Especially for large pull requests this makes testing a lot easier.
+* Actively look for reviews. Do not just open the pull request and wait.
+
+Reviewer
+---------
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:64]:
```suggestion
Reviewer
^^^^^^^^
```

File: docs/dev/guidelines/development-process.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+* Make sure all steps in the `Checklist <https://github.com/ls1intum/Artemis/blob/develop/.github/PULL_REQUEST_TEMPLATE.md>`_ are completed
+* Add or update the "Steps for Testing" in the description of your pull request
+* Make sure that the changes in the pull request are only the ones necessary
+* Mark the pull request as `ready for review <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request>`_
+
+3. Review
+---------
+
+Developer
+---------
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:59]:
```suggestion
Developer
^^^^^^^^^
```
This should be a sub-sub-section.

File: docs/dev/guidelines/development-process.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    * Make sure that (extensive) comments are present where deemed necessary
+    * Performance is reasonable (e.g. Number of database queries or HTTP calls)
+* Submit your comments and status ((thumbs up) Approve or (thumbs down) Request Changes) using GitHub.
+    * Explain what you did (test, review code) and on which test server in the review comment
+
+4. Respond to review
+--------------------
+
+Developer
+---------
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:79]:
```suggestion
Developer
^^^^^^^^^
```

File: docs/dev/guidelines/development-process.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+* Use the pull request to discuss comments or ask questions
+* Update your code where necessary
+* Revert to Draft if the changes will take a while during which review is not needed/possible
+* Set back to ready for review afterwards
+* Notify the reviewer(s) once your revised version is ready for the next review
+* Comment on "inline comments" (e.g. "Done")
+
+Reviewer
+---------
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:88]:
```suggestion
Reviewer
^^^^^^^^
```

File: docs/dev/guidelines/development-process.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+* Respond to questions raised by the reviewer
+* Mark Conversations as Resolved if the change is sufficient
+
+Iterate steps 3 & 4 until ready for merge (- all reviewers approve (thumbs up) )
+--------------------------------------------------------------------------------
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:93]:
```suggestion
Iterate steps 3 & 4 until ready for merge (- all reviewers approve (thumbs up) )
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/dev/guidelines/development-process.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+******************************************
+
+.. contents:: Content of this document
+    :local:
+    :depth: 2
+
+Naming Conventions for GitHub Pull Requests
+===========================================
+
+1. The first term is a main feature of Artemis and is using code highlighting, e.g.  “``Programming exercises``:”
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:12]:
```suggestion
1. The first term is a main feature of Artemis and is using code highlighting, e.g.  “``Programming exercises``:”.
```

File: docs/dev/guidelines/development-process.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+    1. Possible feature tags are: ``Programming exercises``, ``Modeling exercises``, ``Text exercises``, ``Quiz exercises``, ``File upload exercises``, ``Lectures``, ``Exam mode``, ``Assessments``, ``Discussion``, ``Notifications``. More tags are possible if they make sense.
+    2. If no feature makes sense, and it is a pure development or test improvement, we use the term “``Development``:”. More tags are also possible if they make sense.
+    3. Everything else belongs to the ``General`` category.
+
+2. The colon is not highlighted
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:18]:
```suggestion
2. The colon is not highlighted.
```

File: docs/dev/guidelines/development-process.rst [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+Steps to Create and Merge a Pull Request
+========================================
+
+0. Precondition -> only Developer
+---------------------------------
+
+* Limit yourself to one functionality per pull request
+* Split up your task in multiple branches & pull requests if necessary
+* `Commit Early, Commit Often, Perfect Later, Publish Once <https://speakerdeck.com/lemiorhan/10-git-anti-patterns-you-should-be-aware-of>`_
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:40]:
```suggestion
* Limit yourself to one functionality per pull request.
* Split up your task in multiple branches & pull requests if necessary.
* `Commit Early, Commit Often, Perfect Later, Publish Once. <https://speakerdeck.com/lemiorhan/10-git-anti-patterns-you-should-be-aware-of>`_
```

File: docs/dev/guidelines/development-process.rst [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+1. Start Implementation -> only Developer
+-----------------------------------------
+
+* `Open draft pull request. <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request>`_ Allows for code related questions and discussions
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:45]:
```suggestion
* `Open a draft pull request. <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request>`_ This allows for code related questions and discussions.
```

File: docs/dev/guidelines/development-process.rst [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+2. Implementation is "done" -> only Developer
+---------------------------------------------
+
+* Make sure all steps in the `Checklist <https://github.com/ls1intum/Artemis/blob/develop/.github/PULL_REQUEST_TEMPLATE.md>`_ are completed
+* Add or update the "Steps for Testing" in the description of your pull request
+* Make sure that the changes in the pull request are only the ones necessary
+* Mark the pull request as `ready for review <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request>`_
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:53]:
```suggestion
* Make sure all steps in the `Checklist <https://github.com/ls1intum/Artemis/blob/develop/.github/PULL_REQUEST_TEMPLATE.md>`_ are completed.
* Add or update the "Steps for Testing" in the description of your pull request.
* Make sure that the changes in the pull request are only the ones necessary.
* Mark the pull request as `ready for review. <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request>`_
```

File: docs/dev/guidelines/development-process.rst [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+Reviewer
+---------
+* Perform the "Steps for Testing" and verify that the new functionality is working as expected.
+* Verify that related functionality is still working as expected.
+* Check the Changes to
+    * Conform with the code style
+    * Make sure you can easily understand the code
+    * Make sure that (extensive) comments are present where deemed necessary
+    * Performance is reasonable (e.g. Number of database queries or HTTP calls)
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:71]:
```suggestion
    * conform with the code style.
    * make sure you can easily understand the code.
    * make sure that (extensive) comments are present where deemed necessary.
    * performance is reasonable (e.g. number of database queries or HTTP calls).
```
The continuation of a sentence should start with a lowercase letter.

File: docs/dev/guidelines/development-process.rst [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+Developer
+---------
+* Organize or join a testing session. Especially for large pull requests this makes testing a lot easier.
+* Actively look for reviews. Do not just open the pull request and wait.
+
+Reviewer
+---------
+* Perform the "Steps for Testing" and verify that the new functionality is working as expected.
+* Verify that related functionality is still working as expected.
+* Check the Changes to
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:67]:
```suggestion
* Check the changes to
```

File: docs/dev/guidelines/development-process.rst [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+    * Conform with the code style
+    * Make sure you can easily understand the code
+    * Make sure that (extensive) comments are present where deemed necessary
+    * Performance is reasonable (e.g. Number of database queries or HTTP calls)
+* Submit your comments and status ((thumbs up) Approve or (thumbs down) Request Changes) using GitHub.
+    * Explain what you did (test, review code) and on which test server in the review comment
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:73]:
```suggestion
    * Explain what you did (test, review code) and on which test server in the review comment.
```

File: docs/dev/guidelines/development-process.rst [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+--------------------
+
+Developer
+---------
+* Use the pull request to discuss comments or ask questions
+* Update your code where necessary
+* Revert to Draft if the changes will take a while during which review is not needed/possible
+* Set back to ready for review afterwards
+* Notify the reviewer(s) once your revised version is ready for the next review
+* Comment on "inline comments" (e.g. "Done")
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:85]:
```suggestion
* Use the pull request to discuss comments or ask questions.
* Update your code where necessary.
* Revert to draft if the changes will take a while during which review is not needed/possible.
* Set back to ready for review afterwards.
* Notify the reviewer(s) once your revised version is ready for the next review.
* Comment on "inline comments" (e.g. "Done").
```

File: docs/dev/guidelines/development-process.rst [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+
+Reviewer
+---------
+* Respond to questions raised by the reviewer
+* Mark Conversations as Resolved if the change is sufficient
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:90]:
```suggestion
* Respond to questions raised by the reviewer.
* Mark conversations as resolved if the change is sufficient.
```

File: docs/dev/guidelines/development-process.rst [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+
+Iterate steps 3 & 4 until ready for merge (- all reviewers approve (thumbs up) )
+--------------------------------------------------------------------------------
+
+5. Merge
+--------
+A Project Maintainer merges your changes to develop
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:97]:
```suggestion
A project maintainer merges your changes into the ``develop`` branch.
```

File: docs/dev/guidelines/development-process.rst [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+    1. Possible feature tags are: ``Programming exercises``, ``Modeling exercises``, ``Text exercises``, ``Quiz exercises``, ``File upload exercises``, ``Lectures``, ``Exam mode``, ``Assessments``, ``Discussion``, ``Notifications``. More tags are possible if they make sense.
+    2. If no feature makes sense, and it is a pure development or test improvement, we use the term “``Development``:”. More tags are also possible if they make sense.
+    3. Everything else belongs to the ``General`` category.
+
+2. The colon is not highlighted
+
+3. After the colon, there should be a verbal form that is understandable by end users and non-technical persons, because this will automatically become part of the release notes.
+
+    1. The text should be short, non-capitalized (except the first word) and should include the most important keywords. Do not repeat the feature if it is possible.
+    2. We generally distinguish between Bugfixes (the verb “Fix”) and Improvements (all kinds of verbs) in the release notes. This should be immediately clear from the title.
```
Comment (b-fein) [inline docs/dev/guidelines/development-process.rst:23]:
```suggestion
    2. We generally distinguish between bugfixes (the verb “Fix”) and improvements (all kinds of verbs) in the release notes. This should be immediately clear from the title.
```

**Review by b-fein — APPROVED** [review]
Sorry for bombarding you with inline comments. They are mostly about unifying the punctuation and capitalisation.

The new parts of the documentation look good otherwise.

**Comment by MDK5** [comment]
@b-fein thanks for going through it all!

**Review by b-fein — APPROVED** [review]
Thanks! Looks good now.

**Review by Olimbia23 — APPROVED** [review]
Guidelines look good.
Nice work @MDK5
Approve 👍

**Review by krusche — APPROVED** [review]

**Pull request merged**